### DISCUSSION TITLE
[Berry] Clear timer?

### DISCUSSION THREAD

**Discussion by Slyke** [arendst/Tasmota#16704] [Q&A] [answered]
Is there any way to clear a timer once set in Berry?

For example:

```
def t() print("Booh!") end
var myTimer = tasmota.set_timer(5000, t)
myTimer.clear()
// Nothing happens because the timer was cleared
```

https://tasmota.github.io/docs/Berry/#timers

My goal is to send an MQTT payload 10 seconds after a button was last pressed. If it's pressed before the 10 seconds has passed, the countdown resets. I could just make it decrement every second from 10, and reset it back to 10 if a button was pressed, but was hoping for a more simpler way.

**Comment by s-hadinger** [comment, marked as answer]
No this is not supported. The reason is that the timer only stores the timestamp, not the duration. You need to remove the timer and set a new one. 

You can simulate this behavior with `every_second()` if you prefer

**Reply by sfromis** [reply]
Here's a small helper function doing what you suggest, and more
```
def resetable_timer(millis, f)
  class _resetable_timer
    var millis
    def init(millis)
      self.millis = millis
      self.reset()
    end
    def reset()
      self.clear()
      tasmota.set_timer(self.millis, f, self)
    end
    def clear()
      tasmota.remove_timer(self)
    end
  end
  return _resetable_timer(millis)
end
rt=resetable_timer(6000, /->print('tick'))
...
rt.reset()
```

**Reply by Slyke** [reply]
Thank you! That works just fine.
